File: leetcode/0824-goat-latin/solution.cpp
#include <string>
#include <vector>
#include <sstream>
using namespace std;

class Solution {
public:
    string toGoatLatin(string sentence) {
        string vowels = "aeiouAEIOU";
        vector<string> words;
        stringstream ss(sentence);
        string word;
        while (ss >> word) words.push_back(word);

        string result;
        for (int i = 0; i < (int)words.size(); i++) {
            string w = words[i];
            string newWord;
            if (vowels.find(w[0]) != string::npos) {
                newWord = w + "ma";
            } else {
                newWord = w.substr(1) + w[0] + "ma";
            }
            newWord += string(i + 1, 'a');

            if (i > 0) result += ' ';
            result += newWord;
        }

        return result;
    }
};

File: leetcode/0824-goat-latin/Solution.java
class Solution {
    public String toGoatLatin(String sentence) {
        String vowels = "aeiouAEIOU";
        String[] words = sentence.split(" ");
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < words.length; i++) {
            String word = words[i];
            String newWord;
            if (vowels.indexOf(word.charAt(0)) >= 0) {
                newWord = word + "ma";
            } else {
                newWord = word.substring(1) + word.charAt(0) + "ma";
            }
            StringBuilder wordBuilder = new StringBuilder(newWord);
            for (int k = 0; k <= i; k++) {
                wordBuilder.append('a');
            }
            if (i > 0) sb.append(' ');
            sb.append(wordBuilder);
        }

        return sb.toString();
    }
}
